File: 396. rotate-function.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        n = len(nums)

        total_sum = sum(nums)
        F = sum(i * nums[i] for i in range(n))

        ans = F

        for k in range(1, n):
            F = F + total_sum - n * nums[n - k]
            ans = max(ans, F)

        return ans


# Test cases
if __name__ == "__main__":
    from typing import List
    
    solution = Solution()
    
    # Example 1
    nums1 = [4, 3, 2, 6]
    result1 = solution.maxRotateFunction(nums1)
    print(f"Example 1: Input: nums = {nums1}")
    print(f"Output: {result1}")
    print(f"Expected: 26\n")
    
    # Example 2
    nums2 = [100]
    result2 = solution.maxRotateFunction(nums2)
    print(f"Example 2: Input: nums = {nums2}")
    print(f"Output: {result2}")
    print(f"Expected: 0")

File: 396. rotate-function.java
class Solution {
    public int maxRotateFunction(int[] nums) {
        int n = nums.length;

        long sum = 0;
        long F = 0;

        for (int i = 0; i < n; i++) {
            sum += nums[i];
            F += (long) i * nums[i];
        }

        long ans = F;

        for (int k = 1; k < n; k++) {
            F = F + sum - (long) n * nums[n - k];
            ans = Math.max(ans, F);
        }

        return (int) ans;
    }
    
    // Test cases
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Example 1
        int[] nums1 = {4, 3, 2, 6};
        int result1 = solution.maxRotateFunction(nums1);
        System.out.println("Example 1: Input: nums = [4, 3, 2, 6]");
        System.out.println("Output: " + result1);
        System.out.println("Expected: 26\n");
        
        // Example 2
        int[] nums2 = {100};
        int result2 = solution.maxRotateFunction(nums2);
        System.out.println("Example 2: Input: nums = [100]");
        System.out.println("Output: " + result2);
        System.out.println("Expected: 0");
    }
}